File: 242-valid-anagram/valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        unordered_map<char, int> umap;

        int n = s.length();
        int m= t.length();

        if(n != m)return false;

        for(int i=0;i<n;i++)
        {
            umap[s[i]]++;
        }

        for(auto it: t)
        {
            umap[it]--;

        }

        for (auto it : umap) {
            if (it.second != 0) return false;  // <-- check frequencies
        }

        
        return true;
    }
};

File: 242-valid-anagram/valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        if(s.length() != t.length())return false;


        HashMap<Character, Integer> umap = new HashMap<>();

        for(int i=0;i<s.length();i++)
        {
            char ch = s.charAt(i);
            umap.put(ch, umap.getOrDefault(ch, 0) + 1);
        }
        
        for(int i=0;i<t.length();i++)
        {
            char ch = t.charAt(i);

            umap.put(ch, umap.getOrDefault(ch,0) - 1);
        }

        for (int value : umap.values()) {
            if (value != 0) return false;
        }
        return true;
    }
}